### DISCUSSION TITLE
Bug: Texture blurry after scale when sourced from spritesheet.

### DISCUSSION THREAD

**Discussion by JakeHaitsma** [pixijs/pixijs#9807] [Q&A] [answered]
### Current Behavior

A sprite created from a texture created via a spritesheet blurs significantly when scaled up. When loading the same individual image directly, no such blur occurs. See attached screenshot (left is pixel loaded directly, right is pixel from spritesheet).

<img width="213" alt="Screenshot 2023-10-30 at 1 53 40 AM" src="https://github.com/pixijs/pixijs/assets/6137499/990fc408-04b5-41c3-adc0-96ccb7758a00">

### Expected Behavior

Sprites scale the same regardless of texture source.

### Steps to Reproduce

https://codesandbox.io/s/bitter-surf-mf9xpz?file=/spritesheet.json

Load a 2x2 spritesheet with a black pixel in the bottom right corner. Add a single spritesheet frame pointing to that 1x1 pixel.
Load a 1x1 black pixel.
Create new sprites from both textures. Scale to 100x100.

The pixel loaded directly covers the entire container with black. The pixel loaded from a spritesheet is blurry.

### Environment

- **`pixi.js` version**: 7.3.2
- **Browser & Version**: Brave, Version 1.59.124 Chromium: 118.0.5993.117 (Official Build) (x86_64)
- **OS & Version**: macOS 14.0
- **Running Example**: https://codesandbox.io/s/bitter-surf-mf9xpz?file=/spritesheet.json

### Additional Information

Reproduced behavior on Safari Version 17.0 (19616.1.27.211.1).

**Comment by jakerdy** [comment, marked as answer]
> 2x2 spritesheet

Spritesheets has limitations, due to their nature.

Not a bug actually. You must add padding between sprites inside your atlas (2px at least). Otherwice sprites will "leak" onto each other. Because of texture filtering, which interpolates two adjascent pixels in your atlas. This is also truth for other engines, because texture filtering works the same evrywhere.

As a workarround you could switch off texture filtering for atlas texture. If for some reason you have to use 1x1 altas pixels. But this has it's own implications, like precesion issues when you could see some random pixels from neigbour pixels in atlas.

In general, if you want to use sprites for coloring rectangles, make them bigger, like 4x4, and use it's 2x2 center part, and don't turn on mipmaps for that atlas. This will prevent color leakage. In terms of size efficiency, I bet that png with solid color rects wouldn't be that much bigger, maybe couple dozen bytes or so.
